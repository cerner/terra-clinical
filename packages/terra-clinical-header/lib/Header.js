'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _terraArrange = require('terra-arrange');

var _terraArrange2 = _interopRequireDefault(_terraArrange);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

require('terra-base/lib/baseStyles');

require('./Header.scss');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var propTypes = {
  /*
   * Content to be displayed at the start of the header, placed before the title
   */
  startContent: _react.PropTypes.element,

  /*
   * Text to be displayed as the title in the header bar
   */
  title: _react.PropTypes.string,

  /*
   * Content to be displayed at the end of the header
   */
  endContent: _react.PropTypes.element
};

var defaultProps = {
  title: '',
  startContent: null,
  endContent: null
};

var Header = function Header(_ref) {
  var title = _ref.title,
      startContent = _ref.startContent,
      endContent = _ref.endContent,
      customProps = _objectWithoutProperties(_ref, ['title', 'startContent', 'endContent']);

  var attributes = _extends({}, customProps);
  var headerClassNames = (0, _classnames2.default)(['terraClinical-Header', attributes.className]);

  var titleElement = _react2.default.createElement(
    'h1',
    { className: 'terraClinical-Header-title' },
    title
  );
  var headerContent = void 0;

  if (startContent || endContent) {
    headerContent = _react2.default.createElement(_terraArrange2.default, {
      fitStart: startContent,
      fitEnd: endContent,
      fill: titleElement,
      align: 'center'
    });
  } else {
    headerContent = titleElement;
  }

  return _react2.default.createElement(
    'header',
    _extends({}, attributes, { className: headerClassNames }),
    headerContent
  );
};

Header.propTypes = propTypes;
Header.defaultProps = defaultProps;

exports.default = Header;