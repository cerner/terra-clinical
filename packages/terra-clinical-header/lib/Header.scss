@import '~terra-mixins';

.terraClinical-Header {
  background-color: $terra-grey-5;
  border-bottom-color: $terra-grey-15;
  border-bottom-style: solid;
  border-bottom-width: 1px;
  height: 3.587rem;
  padding: 0.714rem;
  width: 100%;

  .terraClinical-Header-title {
    font-size: 1.285rem;
    font-weight: normal;
    line-height: 1.75;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
    word-break: normal;
    word-wrap: normal;
  }
}
