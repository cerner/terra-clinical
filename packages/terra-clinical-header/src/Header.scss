:local {
  .flex-header {
    background-color: #f4f4f4;
    border-bottom-color: #dedfe0;
    border-bottom-style: solid;
    border-bottom-width: 1px;
    display: flex;
    justify-content: space-between;
    min-height: 3.587rem;
    overflow: hidden;
    padding: 0.714rem;
    position: relative;
    width: 100%;
  }

  .flex-fill {
    align-self: center;
    flex: 1 1 auto;
    height: 100%;
    min-width: 0;
    position: relative;
    width: 100%;
  }

  .flex-end {
    align-self: center;
    display: flex;
    flex: 0 0 auto;
    height: 100%;
    position: relative;
  }

  .flex-end > * {
    display: flex;
    width: 100%;
  }

  .title-container {
    height: 100%;
    left: 0;
    position: inherit;
    top: 0;
    width: 100%;
  }

  .title {
    font-size: 1.285rem;
    font-weight: normal;
    height: 100%;
    line-height: 1.75;
    margin-bottom: 0;
    margin-top: 0;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: normal;
    width: 100%;
    word-break: break-word;
    word-wrap: break-word;
  }

  .flex-subheader {
    background-color: #c8cacb;
    border-bottom-color: initial;
    border-bottom-style: 0;
    border-bottom-width: 0;
    display: flex;
    height: 2.857rem;
    justify-content: space-between;
    overflow: hidden;
    padding: 0.3571rem 0.7143rem;
    position: relative;
    width: 100%;
  }
}
