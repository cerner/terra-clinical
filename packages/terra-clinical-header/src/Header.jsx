import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import 'terra-base/lib/baseStyles';
import styles from './Header.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * Content to be displayed at the start of the header, placed before the title
   */
  startContent: PropTypes.element,

  /**
   * Text to be displayed as the title in the header bar
   */
  title: PropTypes.string,

  /**
   * Content to be displayed at the end of the header
   */
  endContent: PropTypes.element,

  /**
   * A Boolean indicating if element is a subheader.
   */
  isSubheader: PropTypes.bool,
};

const defaultProps = {
  title: '',
  startContent: null,
  endContent: null,
  isSubheader: false,
};

const Header = ({ title, startContent, endContent, isSubheader, ...customProps }) => {
  let startElement;
  if (startContent) {
    startElement = <div className={cx('flexEnd')}>{startContent}</div>;
  }

  let titleElement;
  if (title) {
    titleElement = (
      <div className={cx('titleContainer')}>
        <h1 className={cx('title')}>
          {title}
        </h1>
      </div>
    );
  }

  let endElement;
  if (endContent) {
    endElement = <div className={cx('flexEnd')}>{endContent}</div>;
  }

  const headerClass = isSubheader ? 'flexSubheader' : 'flexHeader';

  return (
    <header {...customProps} className={cx(headerClass, customProps.className)}>
      {startElement}
      <div className={cx('flexFill')}>
        {titleElement}
      </div>
      {endElement}
    </header>
  );
};

Header.propTypes = propTypes;
Header.defaultProps = defaultProps;

export default Header;
