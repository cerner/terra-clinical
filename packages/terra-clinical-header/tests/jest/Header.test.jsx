import React from 'react';
import Header from '../../src/Header';

it('should render a default component', () => {
  const header = render(<Header />);
  expect(header).toMatchSnapshot();
});

it('should render a header with title only', () => {
  const header = render(<Header title="title" />);
  expect(header).toMatchSnapshot();
});

it('should render a header with content on the left', () => {
  const header = render(<Header startContent={<div>start content</div>} />);
  expect(header).toMatchSnapshot();
});

it('should render a header with content on the right', () => {
  const header = render(<Header endContent={<div>end content</div>} />);
  expect(header).toMatchSnapshot();
});

it('should render a header with all content', () => {
  const header = render((
    <Header
      startContent={<div>start content</div>}
      title="Title"
      endContent={<div>end content</div>}
    />
  ));
  expect(header).toMatchSnapshot();
});

it('should render a subheader with title only', () => {
  const subheader = render(<Header title="title" isSubheader />);
  expect(subheader).toMatchSnapshot();
});

it('should render a subheader with all content', () => {
  const subheader = render((
    <Header
      startContent={<div>start content</div>}
      title="Title"
      endContent={<div>end content</div>}
      isSubheader
    />
  ));
  expect(subheader).toMatchSnapshot();
});
