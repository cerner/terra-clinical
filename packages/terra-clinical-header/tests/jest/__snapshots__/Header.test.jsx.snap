exports[`test should render a default component 1`] = `
<header
  class="flexHeader" />
`;

exports[`test should render a header with all content 1`] = `
<header
  class="flexHeader">
  <div
    class="flexEnd">
    <div>
      start content
    </div>
  </div>
  <div
    class="flexFill">
    <div
      class="titleContainer">
      <h1
        class="title">
        Title
      </h1>
    </div>
  </div>
  <div
    class="flexEnd">
    <div>
      end content
    </div>
  </div>
</header>
`;

exports[`test should render a header with content on the left 1`] = `
<header
  class="flexHeader">
  <div
    class="flexEnd">
    <div>
      start content
    </div>
  </div>
</header>
`;

exports[`test should render a header with content on the right 1`] = `
<header
  class="flexHeader">
  <div
    class="flexEnd">
    <div>
      end content
    </div>
  </div>
</header>
`;

exports[`test should render a header with title only 1`] = `
<header
  class="flexHeader">
  <div
    class="flexFill">
    <div
      class="titleContainer">
      <h1
        class="title">
        title
      </h1>
    </div>
  </div>
</header>
`;

exports[`test should render a subheader with all content 1`] = `
<header
  class="flexSubheader">
  <div
    class="flexEnd">
    <div>
      start content
    </div>
  </div>
  <div
    class="flexFill">
    <div
      class="titleContainer">
      <h1
        class="title">
        Title
      </h1>
    </div>
  </div>
  <div
    class="flexEnd">
    <div>
      end content
    </div>
  </div>
</header>
`;

exports[`test should render a subheader with title only 1`] = `
<header
  class="flexSubheader">
  <div
    class="flexFill">
    <div
      class="titleContainer">
      <h1
        class="title">
        title
      </h1>
    </div>
  </div>
</header>
`;
